### PR TITLE
handle last empty column (fix to master)

### PR DESCRIPTION
**Pull request by Lagovas**
fix the case when last data in data row column from PostgreSQL come with empty data and server try to read 0 bytes, take io.EOF and stop processing connection due to an unexpected error in parsing packets

### CODE REVIEW

File: decryptor/postgresql/pg_escape_decryptor.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (decryptor *PgEscapeDecryptor) readDataLength(reader io.Reader) (uint64, []
 		return 0, decryptor.octLengthBuf[:octLenCount], base.ErrFakeAcraStruct
 	}
 	decryptor.outputSize += octLenCount
-	binary.Read(bytes.NewBuffer(decryptor.lengthBuf[:]), binary.LittleEndian, &length)
+	binary.Read(bytes.NewReader(decryptor.lengthBuf[:]), binary.LittleEndian, &length)
```
Comment (vixentael) [inline decryptor/postgresql/pg_escape_decryptor.go]: what's the difference?

**Comment by vixentael** [comment]
Same as #315 but for master branch

File: decryptor/postgresql/pg_escape_decryptor.go (hunk 1 above)
Comment (Lagovas) [reply decryptor/postgresql/pg_escape_decryptor.go]: bytes.Reader wrap []byte with method Read to implement io.Reader and just return data from bytes
bytes.NewBuffer has overhead with internal buffer for short slices and store []byte too. NewReader doesn't have extra internal buffer and uses less memory

Comment (vixentael) [reply decryptor/postgresql/pg_escape_decryptor.go]: > uses less memory

👍

**Pull request merged**